### PR TITLE
[cluster-api] Handle ignored errors

### PR DESCRIPTION
**Pull request by mboersma**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

Cleans up code in the cluster-api cluster-autoscaler backend, in particular handling ignored errors. Also makes other small changes suggested by golangci-lint.

#### Which issue(s) this PR fixes:

N/A

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

File: cluster-autoscaler/cloudprovider/clusterapi/clusterapi_controller_test.go [hunk 1]
```diff
@@ -385,7 +383,9 @@ func createTestConfigs(specs ...testSpec) []*testConfig {
 			}
 			config.machineDeployment.SetAnnotations(spec.annotations)
 			config.machineDeployment.SetLabels(machineDeploymentLabels)
-			unstructured.SetNestedStringMap(config.machineDeployment.Object, machineDeploymentLabels, "spec", "selector", "matchLabels")
+			if err := unstructured.SetNestedStringMap(config.machineDeployment.Object, machineDeploymentLabels, "spec", "selector", "matchLabels"); err != nil {
+				panic(err)
```
Comment (mboersma) [inline cluster-autoscaler/cloudprovider/clusterapi/clusterapi_controller_test.go:387]: I wagered that using `panic()` in a unit test might be acceptable. It's either that or change several functions in this unit test to return and expect errors; I'm happy to do that if it's preferable.

File: cluster-autoscaler/cloudprovider/clusterapi/clusterapi_controller_test.go [hunk 2]
```diff
@@ -1899,11 +1903,6 @@ func Test_machineController_nodeGroupForNode(t *testing.T) {
 	}, nil)
 	mdTestConfigs = append(mdTestConfigs, uniqueMDConfig)
 
-	allMachineDeployments := make([]*unstructured.Unstructured, 0, len(mdTestConfigs))
```
Comment (mboersma) [inline cluster-autoscaler/cloudprovider/clusterapi/clusterapi_controller_test.go:1902]: This and the similar section below look like they were copied from a similar unit test, but the values aren't used here.

File: cluster-autoscaler/cloudprovider/clusterapi/clusterapi_controller_test.go (hunk 1 above)
Comment (elmiko) [reply cluster-autoscaler/cloudprovider/clusterapi/clusterapi_controller_test.go:387]: i think this is probably ok

**Review by elmiko — COMMENTED** [review]
this is nice cleanup, thanks @mboersma !

/approve
/lgtm

**Pull request merged**